Docs 7726 update mdip onboarding details pa (#224)
mulesoft/docs-api-manager: modules/ROOT/pages/configure-multiple-credential-providers.adoc

== The Use Case

> To better understand how to implement multiple external client providers, imagine a company, "Good Weather" that uses meteorology to forecast changing weather. Good Weather collects quantitative data about the current state of the atmosphere at a given place and, using meteorology, predicts how that current state will change over time.

Reviewer: Good Weather should be fully parenthetical (two commas) or not (no commas):
```suggestion
To better understand how to implement multiple external client providers, imagine a company named "Good Weather" that uses meteorology to forecast changing weather. Good Weather collects quantitative data about the current state of the atmosphere at a given place and, using meteorology, predicts how that current state will change over time.
```

Author: Done! Thanks!

Good Weather has several sub-organizations running under the master organization. These sub-organizations exist based on the functions they perform, for example, Engineering and Forecasting. The Forecasting department researches and collects recent meterological data from various sources. This data is then consumed by various client applications. While the Engineering department maintains the applications that host the forecasting research data.

Because Good Weather must ensure that its weather-forecasting data is accurate and secure, the weather APIs are accessed only by employees of the Forecasting department. Good Weather also wants to ensure that the production APIs are the single source of truth at all times.

To ensure security and access segregation for different organizations, environments, and APIs, Good Weather IT sets up separate external client providers for the Forecasting department.

== Onboarding Multiple Client Providers

> Your onbaording multiple client providers procedure differs based on how you have configured client providers and your business organizations in your company:

Reviewer: ```suggestion
The procedure that you use to onboard multiple client providers differs based on how you have configured client providers and the business organizations in your company:
```

Author: Done! Thanks!

> * *Scenario 1: No external providers configured:*

Reviewer: This seems to be an inline heading (bold) that also follows bullet list item style (sentence).

We really do not have a sanctioned format or process for inline headings, so I suggest that you make these bullet-list items, without the boldface and without the colon:

```suggestion
* No external client providers configured
```

Author: Done! Thanks!

+

> Creating and modifying such environments and APIs behave the same. The native Anypoint Platform client provider is assigned for such environments. The existing APIs and environment have no impact.

Reviewer: To what does "behave the same" refer: creating and modifying? Environments and APIs? something else?
Please reword to clarify: Both creating and modifying involve the same steps? the results are the same? Or creating and modifying is the same for environments and APIs?

Reviewer: Or do you intend to say that creating and modifying environments and APIs is not affected by / has no effect on existing ones (which cannot be true in the case of modifying...).

Author: Done! Thanks!

> * *Scenario 2: External client provider configured:*

Reviewer: ditto
```suggestion
* External client providers configured
```

Author: Done! Thanks!

+ 

> If you have external client providers, such as PingFederate, configured for your environments, the client provider is automatically ported when you onboard the multiple client providers feature. However, when you create a new environment, you must manually assign the correct client provider to your environment in Access Management. 

Reviewer: ```suggestion
If you have external client providers such as PingFederate configured for your environments, the client provider is automatically ported when you onboard the multiple client providers feature. However, when you create a new environment, you must manually assign the correct client provider to your environment in Access Management. 
```
HOWEVER...are you onboarding the FEATURE or the actual client providers?

Author: No the feature. Thanks! Done!

> *Best Practice*

Reviewer: Does this apply to one or both of the two environments listed? Clarify, but don't make this text with a heading, but just go straight into a new para.

Author: Environment here means something totally different. This is a best practice of how to implement MiDP period. Doesn't matter what the user's current setup is.

+

> To secure your APIs by leveraging the multiple client providers feature, create one external client provider per environment. Assign the existing client provider to your production environment, while adding the new providers to other environments, such as sandbox, and QA. 

Reviewer: ```suggestion
To secure your APIs, the best practice is to create one external client provider per environment. Assign the existing client provider to your production environment when you add new providers to other environments, such as the sandbox, or when you perform QA. 
```
I assumed that "and QA" meant this, but perhaps you mean the QA environment? Clarify, if so.

Author: You can secure your APIs in many different ways, such as by applying policies and such. Just mentioning "To secure your APIs" makes one feel that is the only way to secure your API. And plus the title of the section is already Best Practice. QA is also an environment like production.

> However, before changing any existing client providers, ensure that you check the impact on managed APIs and their consumers.

Reviewer: Will this cause any confusion with the opening text about new things not affecting existing things?

Reviewer 2: Maybe we need to reference them to the guidelines section.

Author: No, I think that part is clear. Thanks! I'll add a reference, Tomas!
